CRM/Core/BAO/UFField.php

```diff
@@ -42,6 +42,55 @@ class CRM_Core_BAO_UFField extends CRM_Core_DAO_UFField {
   private static $_contriBatchEntryFields = NULL;
   private static $_memberBatchEntryFields = NULL;
 
+  /**
+   * Create UFField object
+   *
+   * @param array $params
+   *   Array per getfields metadata.
+   *
+   * @throws API_Exception
+   *
+   * @return CRM_Core_BAO_UFField
+   */
+  public static function create(&$params) {
+    // CRM-14756: kind of a hack-ish fix. If the user gives the id, uf_group_id is retrieved and then set.
+    if (isset($params['id'])) {
+      $groupId = civicrm_api3('UFField', 'getvalue', array(
+        'return' => 'uf_group_id',
+        'id' => $params['id'],
+      ));
+    }
+    else {
+      $groupId = CRM_Utils_Array::value('uf_group_id', $params);
+    }
+
+    $field_name = CRM_Utils_Array::value('field_name', $params);
+
+    if (strpos($field_name, 'formatting') !== 0 && !CRM_Core_BAO_UFField::isValidFieldName($field_name)) {
+      throw new API_Exception('The field_name is not valid');
```

Comment:
At the BAO level we normally through CRM_Core_Exceptions rather than API_Exceptions